assets/js/base/context/providers/store-notices-with-snackbar/context.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import {
+	createContext,
+	useContext,
+	useCallback,
+	useState,
+} from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import SnackbarNoticesContainer from '@woocommerce/base-context/providers/store-notices-with-snackbar/components/snackbar-notices-container';
+
+/**
+ * Internal dependencies
+ */
+import { useStoreEvents } from '../../hooks/use-store-events';
+import { useEditorContext } from '../editor-context';
+import StoreNoticesContainer from '../store-notices/components/store-notices-container';
+
+/**
+ * @typedef {import('@woocommerce/type-defs/contexts').NoticeContext} NoticeContext
+ * @typedef {import('react')} React
+ */
+
+const StoreNoticesWithSnackbarContext = createContext( {
+	notices: [],
+	createNotice: ( status, text, props ) => void { status, text, props },
+	createSnackbarNotice: ( content, options ) => void { content, options },
+	removeNotice: ( id, ctxt ) => void { id, ctxt },
+	setIsSuppressed: ( val ) => void { val },
+	context: 'wc/core',
+} );
+
+/**
+ * Returns the notices context values.
+ *
+ * @return {NoticeContext} The notice context value from the notice context.
+ */
+export const useStoreNoticesWithSnackbarContext = () => {
+	return useContext( StoreNoticesWithSnackbarContext );
+};
+
+/**
+ * Provides an interface for blocks to add notices to the frontend UI.
+ *
+ * Statuses map to https://github.com/WordPress/gutenberg/tree/master/packages/components/src/notice
+ *  - Default (no status)
+ *  - Error
+ *  - Warning
+ *  - Info
+ *  - Success
+ *
+ * @param {Object} props Incoming props for the component.
+ * @param {React.ReactChildren} props.children The Elements wrapped by this component.
+ * @param {string} props.className CSS class used.
+ * @param {boolean} props.createNoticeContainer Whether to create a notice container or not.
+ * @param {string} props.context The notice context for notices being rendered.
+ */
+export const StoreNoticesWithSnackbarProvider = ( {
+	children,
+	className = '',
+	createNoticeContainer = true,
+	context = 'wc/core',
+} ) => {
+	const { createNotice, removeNotice } = useDispatch( 'core/notices' );
+	const [ isSuppressed, setIsSuppressed ] = useState( false );
+	const { dispatchStoreEvent } = useStoreEvents();
+	const { isEditor } = useEditorContext();
+
+	const createNoticeWithContext = useCallback(
```

Comment:
I think this separate Provider is in the right direction, but by keeping the default notifications part we end up having duplicated logic, which is not ideal. I am thinking to separate the implementations, have a provider just for default notices and one for the snackbar notices. This way there would be no confusion on what we should use, we would not have duplicated logic and the implementations would be decoupled; tests would just focus on the specific type of notification (currently the test for this also tests default notifications, which is already tested in the intial provider).

Reply:
I should have submitted this as a comment, not as a "request changes".